src/components/pages/ChooseSeatsPage.vue

```diff
@@ -1,50 +1,145 @@
-<script>
-import { defineComponent } from 'vue';
+<script setup>
+import { ref, computed, onBeforeMount } from 'vue';
+import { useRouter } from "vue-router";
+import { useMoviesStore } from '@/stores/moviesStore';
+import { getScreeningById } from '@/api/service/Screenings';
+import SectionTitle from '@/components/common/Section/SectionTitle.vue';
+import TheContainer from '@/components/common/TheContainer.vue';
+import ScreeningsCard from '@/components/views/ScreeningsCard.vue';
 import AppButton from '@/components/common/App/AppButton.vue';
-import { useAuthStore } from '@/stores/authStore';
-
-export default defineComponent({
-  setup() {
-    const auth = useAuthStore();
-    return { auth };
-  },
-  components: { AppButton },
-  computed: {
-    isLoggedIn() {
-      return this.auth.isLoggedIn
-    }
-  },
+import BreadCrumbs from '@/components/common/BreadCrumbs.vue';
+import ProgressStep from '@/components/common/ProgressStep.vue';
+import CinemaHall from '@/components/views/CinemaHall.vue';
+
+const props = defineProps({
+  screeningId: {
+    type: Number,
+    required: true
+  }
+})
+
+const router = useRouter();
+
+const moviesStore = useMoviesStore()
+const movieData = computed(() => moviesStore.getMovieByMovieId(screeningData.value.movie))
+
+const screeningData = ref({});
+const screeningDate = computed(() =>
+  screeningData.value.datetime
+    .substring(0, 10)
+    .split('-')
+    .reverse()
+    .join("/")
+)
+const screeningTime = computed(() => screeningData.value.datetime.substring(11,16))
+const screeningDayName = computed(() =>
+  new Date(screeningData.value.datetime)
+    .toLocaleString('en-GB', {weekday: 'long'})
+)
+
+const seatsTaken = computed(() => screeningData.value.taken_seats)
+const seatsAvailable = computed(() => screeningData.value.available_seats)
+
+const allSeats = computed(() =>
+  seatsTaken.value ? seatsTaken.value.concat(seatsAvailable.value) : seatsAvailable.value)
+
+const lastSeat = computed(() => {
+  const threeCharsLabels = allSeats.value.filter(seat => seat.length === 3)
+  const sortedLabels = threeCharsLabels.sort()
+  return sortedLabels.slice(-1)
+})
+
+const rowsNumber = computed(() => {
+  const letter = lastSeat.value[0].charAt(0).toLowerCase()
+  const alphabet = 'abcdefghijklmnopqrstuvwxyz'
+  return alphabet.indexOf(letter) + 1
+})
+
+const seatsInRow = computed(() => parseInt(lastSeat.value[0].slice(1, 3)))
+
+const isRequestFinished = ref(false)
```

Comment:
if you initialize `screeningData` as `null` you may not need this additional ref and just check `!!screeningData.value` - single source of truth is always a good choice :)